### ISSUE TITLE
Incorporating react-digraph

### ISSUE THREAD

**Issue by dbvrac** [chentsulin/electron-react-boilerplate#1758]
My v0.13.3 app uses react-digraph, but rather than importing it as a dependency in package.json I have the files in my /app/components/ directory. I did this because I'm customizing it heavily, though I realize this may be the root of my troubles:
I'm stuck, trying to figure out why react-digraph seems to work just fine when run with `npm run dev` but only gives blue screen when I build or package (Mac and Linux). I ran with `DEBUG_PROD=true yarn run start` and saw in the console `Uncaught ReferenceError: Graph is not defined`. I made a branch and ran some of the various tests and fixes listed under "scripts" in package.json, but I'm very weak on interpreting the output. Eventually, running `DEBUG_PROD=true yarn run start`, I was able to finally see a graph (though nothing worked), but with console errors starting with `Uncaught TypeError: Cannot read property 'getBBox' of null` and trying any of the graph controls gave me `Uncaught TypeError: Cannot read property 'clientWidth' of null`.

`const viewBBox = entities.getBBox();` and `const width = parent.clientWidth;` are in react-digraph /src/components/graph-view.js.

I intend to upgrade my base project soon, but I want to nail this down first, unless that's not advised.

**Comment by vikr01** [comment]
Could you possibly show a bit of your source code as to how you imported and rendered `Graph` from `react-digraph`? It would be much easier to figure out what the issue is if I can reproduce it myself.

Just to make sure -- you have `d3@^4.1.1` installed right? [It is a peer dependency of `react-digraph`](https://github.com/uber/react-digraph/blob/master/package.json#L28) and [the issue is originating from `d3` returning null where you don't want it to](url).

`this.entities` in `react-digraph` is referring to a `ref` element, which is only created after rendering `GraphView`.

**Comment by dbvrac** [comment]
Yes, `"d3": "^5.5.0",` is in the top-level package.json. It runs just fine with `npm run dev`, but wont package correctly.
The file that imports and renders `GraphView` has the following import statements, and the .js files they point to are in the same directory:
`import GraphView from './graph-view';`
`import GraphConfig from './graph-config';`

Update:
I just ran a test where I put `"react-digraph": "^4.0.3",` in package.json, changed my import statement to `import GraphView from 'react-digraph';`, added 
`const GraphConfig =  { (a bunch of stuff from the react-digraph example) }`, ran `yarn install`, and it packaged just fine.

So maybe at this point I should change my strategy and focus on using react-digraph correctly rather than figuring out why the wrong way doesn't work. `8)` Still, it would be nice to know why it didn't work, for future reference if nothing else.

**Comment by vikr01** [comment]
Not sure why you dropped the `react-digraph` version. The version you're using for `d3` could still cause some issues since it's an unmet peer dependency of `react-digraph`.

I'm still having a hard time understanding how exactly your application is structured without seeing it.

To my understanding --  you've separated your `Graph` class into `./graph-view` and your `GraphConfig` into `./graph-config`?

There could be some issues if you named a component `GraphView` when `react-digraph` exports a `GraphView` class due to `displayName`, but I think this is just for the way components are named in React DevTools.

Since you've separated both files, are you using `props` to pass down the `GraphConfig`?

**Comment by amilajack** [comment]
See this example: https://github.com/amilajack/erb-react-digraph-example

**Comment by dbvrac** [comment]
@vikr01  I updated my package.json to include `"react-digraph": "^4.1.0",` ` "d3": "^4.1.1",` `"react": "^16.4.2",` `"radium": "^0.24.1",` but I'm still getting the same error. I'm narrowing in on the problem you mentioned earlier: "`this.entities` in `react-digraph` is referring to a `ref` element, which is only created after rendering `GraphView`". I'm looking for ways to change the order in which things render, for lack of a better idea.

@amilajack Thanks for that example. I am able to get a generic version of my project to work by using the react-digraph node-module, but I'm looking instead to use my own `graph-view.js` and `graph-config.js` which are imported in my `PlayGround.js` (essentially the same as the graph.js file in the [example](https://github.com/uber/react-digraph/blob/master/src/examples/graph.js)), and my own `graph-controls.js` which is imported by `graph-view.js`. It works when running `npm run dev` but not with `DEBUG_PROD=true yarn run start`, and I am slooooowly understanding why. Here is the error:

> graph-view.js:536 Uncaught TypeError: Cannot read property 'getBBox' of null
    at t.handleZoomToFit (graph-view.js:536)
    at setTimeout (graph-view.js:181)
handleZoomToFit @ graph-view.js:536
setTimeout @ graph-view.js:181
setTimeout (async)
componentDidMount @ graph-view.js:179
Ba @ react-dom.production.min.js:218
La @ react-dom.production.min.js:211
Aa @ react-dom.production.min.js:210
Da @ react-dom.production.min.js:210
Ye @ react-dom.production.min.js:225
Rn @ react-dom.production.min.js:89

I realize I can customize a lot of things in my own `graph-config.js` file, or by using `const GraphConfig = { bla }` in `PlayGround.js`. I've managed to send other props mentioned in the react-digraph [README](https://github.com/uber/react-digraph/blob/master/README.md), such as `maxTitleChars` and I'm passing a tweaked `renderNode` prop, but it needs to send data to my tweaked `renderNodeText` function in `graph-view.js`. I don't (yet) see how I can instead pass this tweak from `graph-controls.js` as a prop to `node-modules/react-digraph/`. 

I anticipate adding a lot of customization, and it seems like using my own files would be easier than sending lots of modifications to the standard `node-modules/react-digraph` package.
So I'm stuck between three choices:
1) Get better at sending props to the standard `node-modules/react-digraph` package.
2) Create my own `node-modules/react-digraph` package that has my tweaked `renderNodeText` function (and probably others not mentioned in the README).
3) Figure out how to get my own custom files to render correctly

I'm ready to be wrong about any of this. 

I hesitate to post my code because a) **it's really ugly**, b) there are several files, including my own `actions` and `reducers`, and c) opening my project to the world would wreak havoc on some API's my project is communicating with.
